src/index.ts

```diff
@@ -1,47 +1,54 @@
 import ejs from "ejs";
-import { getContext, helpers, Property } from "@dylibso/xtp-bindgen";
+import { helpers, getContext, ObjectType, EnumType, ArrayType, XtpNormalizedType, MapType, Parameter, Property } from "@dylibso/xtp-bindgen"
 
-function toGolangType(property: Property): string {
-  if (property.$ref) return goName(property.$ref.name);
-  switch (property.type) {
-    case "string":
-      if (property.format === "date-time") {
-        return "time.Time";
-      }
-      return "string";
-    case "number":
-      if (property.format === "float") {
-        return "float32";
-      }
-      if (property.format === "double") {
-        return "float64";
+function toGolangTypeX(type: XtpNormalizedType, required?: boolean): string {
+  // annotate with null if nullable 
+  const pointerify = (t: string) => {
+    // TODO key off the `type.kind` here instead
+    if (t.startsWith("[]") || t.startsWith("map[")) return t
+    return `${type.nullable || !required ? '*' : ''}${t}`
+  }
+
+  switch (type.kind) {
+    case 'string':
+      return pointerify('string')
+    case 'int32':
+      return pointerify('int32')
+    case 'float':
+      return pointerify('float32')
+    case 'double':
+      return pointerify('float64')
+    case 'byte':
+      return pointerify('byte')
+    case 'date-time':
+      return pointerify('time.Time')
+    case 'boolean':
+      return pointerify('bool')
+    case 'array':
+      const arrayType = type as ArrayType
+      return pointerify(`[]${toGolangTypeX(arrayType.elementType)}`)
+    case 'buffer':
+      return pointerify('[]byte')
+    case 'object':
+      const oType = (type as ObjectType)
+      if (oType.properties?.length > 0) {
+        return pointerify(goName(oType.name))
+      } else {
+        // TODO: shouldn't this be `interface{}` ?
+        return pointerify("map[string]interface{}")
       }
-      return "int64";
-    case "integer":
-      return "int32";
-    case "boolean":
-      return "bool";
-    case "object":
-      return "map[string]interface{}";
```

Comment:
we don't expose `$ref` anymore so now you need to disambiguate what kind of object it is